src/libbson/tests/test-utf8.c

```diff
@@ -66,6 +66,10 @@ test_bson_utf8_escape_for_json (void)
    BSON_ASSERT (0 == memcmp (str, "my\\u0000key", 7));
    bson_free (str);
 
+   str = bson_utf8_escape_for_json ("my\xc0\x80key", 7);
+   BSON_ASSERT (0 == memcmp (str, "my\\u0000key", strlen (str)));
```

Comment:
```suggestion
   ASSERT_CMPSTR (str, "my\\u0000key");
```
Use `ASSERT_CMPSTR` when comparing strings in tests (which includes approprate null checks).

Drive-by update to other assertions in this function/file to use `ASSERT_CMPSTR` is optional.